xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/reference/EntityReference.java

```diff
@@ -42,6 +43,13 @@
  */
 public class EntityReference implements Serializable, Cloneable, Comparable<EntityReference>
 {
+    /**
+     * See {@link #getParentType()}.
+     * @since 17.0.0RC1
+     */
+    @Unstable
+    public static final String FALLBACK_PARENT_TYPE_PARAMETER = "fallbackParentType";
```

Comment:
Feels a bit long, especially when thinking about the serialized reference. What is sure is that it should be consistent with the getter name, which is not the case right now.

Reply:
> What is sure is that it should be consistent with the getter name

It makes sense that we keep having `getParentType`: that method retrieves the parent type based on different info: the parent, and the fallbackParentType parameter. So for me, we shouldn't rename the getter. Now `setParentTypeParameter` should be renamed indeed.

Reply:
Honestly, this "fallback" part does not really make sense to me. It's not "different info", it's different sources to know the parent type: either you set it explicitly or it's deduced from the actual parent or the default parent type mapping. There is no reason to have such a complex name as there is no ambiguity with `parentType` naming, it cannot be anything else.